sdl-core/src/main/scala/io/smartdatalake/util/hdfs/SparkRepartitionDef.scala

```diff
@@ -79,7 +80,29 @@ case class SparkRepartitionDef(numberOfTasksPerPartition: Int,
       repartitionForOnePartitionValue(df) // this is the same as writing only one partition value
     }
     // sort within spark partitions
-    if (sortCols.nonEmpty) dfRepartitioned.sortWithinPartitions(sortCols.map(col):_*)
+    if (sortCols.nonEmpty) {
```

Comment:
Cool.
See the following proposition with Scala Regex Pattern Matching "magic" to simplify the parsing:
```
    if (sortCols.nonEmpty) {
      val sortColDirRegex = "([^\s])\s([^\s])".r
      val sortColRegex = "([^\s])".r
      val sortExprs = sortCols.map {
        case sortColDirRegex(colName, sortDir) if sortDir == "asc" => col(colName).asc
        case sortColDirRegex(colName, sortDir) if sortDir == "desc" => col(colName).desc
        case sortColDirRegex(colName, sortDir) => throw new ConfigurationException(s"""($dataObjectId) Wrong sort direction provided in [sparkRepartition.sortCols] entry "$sortCol". Sort direction must be asc or desc.""", Some("sparkRepartition.sortCols"))
        case sortColRegex(colName) => col(colName)
        case entry => throw new ConfigurationException(s"""($dataObjectId) Too many arguments provided in [sparkRepartition.sortCols] entry "$entry". Just provide colName or colName and sortDir separated by whitespace.""")
      }
      dfRepartitioned.sortWithinPartitions(sortExprs:_*)
    }
```